keanu-project/src/test/java/io/improbable/keanu/network/LambdaSectionTest.java

```diff
@@ -0,0 +1,103 @@
+package io.improbable.keanu.network;
+
+import io.improbable.keanu.vertices.ConstantVertex;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.binary.AdditionVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary.SinVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+import org.junit.Test;
+
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+public class LambdaSectionTest {
+
+    @Test
+    public void doesGetDownstreamProbabilisticVertices() {
+
+        GaussianVertex A = new GaussianVertex(0, 1);
+        GaussianVertex B = new GaussianVertex(0, 1);
+        GaussianVertex C = new GaussianVertex(A.plus(B), 1);
+
+        LambdaSection lambdaSection = LambdaSection.getDownstreamLambdaSection(A, false);
+        Set<Vertex> verticesDepthFirst = lambdaSection.getLatentAndObservedVertices();
+
+        assertEquals(2, verticesDepthFirst.size());
+        assertThat(verticesDepthFirst, containsInAnyOrder(A, C));
```

Comment:
If you can be bothered it may be worth checking the `order` actually matches depth first

Reply:
It also feels a bit odd that the `getDownstream` includes the vertex we called it on? Why is `A` in this?

Reply:
Given this is a hash set - I'm not sure we want to be assuming a particular order?

Reply:
Good point about it being named getVerticesDepthFirst but there's no test to enforce this. I think in general whether it's DFS or BFS the user of this doesn't care. I've changed the name to just getVertices in order to not imply a specific traversal implementation.